### ISSUE TITLE
Memory test report

### ISSUE THREAD

**Issue by yezhem** [TUDB-Labs/multi-lora-fine-tune#66]
can we use this test report in readme @mikecovlee  @merlintang 
![LORA块数量与内存使用关系图](https://github.com/TUDB-Labs/multi-lora-fine-tune/assets/17040545/d4168b8e-aa3c-41a9-bd68-418e07ae4050)

**Comment by merlintang** [comment]
this is good, I think we can put this into readme.

**Comment by merlintang** [comment]
can we have different batch size and sequence length ? meanwhile, we need to explain the sequence length meaning here.

**Comment by yezhem** [comment]
@mikecovlee can add this to readme

**Comment by yezhem** [comment]
> can we have different batch size and sequence length ? meanwhile, we need to explain the sequence length meaning here.

yes, I record the peak memory usage, so the max input data size == micro-batch size * max sequence length. 
The cutoff length equals the max sequence length, no input train data sequence length will exceed the max sequence length, and if exceeded, it will be cut off.

**Comment by merlintang** [comment]
does this micro-batch size = batch size in model pretrain? why not change micro-batch size = 50 and max sequence length=50 instead?

**Comment by yezhem** [comment]
> from the model fine-tune process, we do not have the hyper para: max sequence length. Thus, what is the meaning of max sequence length for a DS?

max sequence length is limited by the llama model and GPU memory. We can set the max sequence length to avoid oom. We also can use the default length like alpaca-lora = 256
micro-batch size does not equal batch size, the batch size also is limited by GPU memory, so we use gradient accumulation to split large batch size into micro-batch size.

**Comment by merlintang** [comment]
> > from the model fine-tune process, we do not have the hyper para: max sequence length. Thus, what is the meaning of max sequence length for a DS?
> 
> max sequence length is limited by the llama model and GPU memory. We can set the max sequence length to avoid oom. We also can use the default length like alpaca-lora = 256 micro-batch size does not equal batch size, the batch size also is limited by GPU memory, so we use gradient accumulation to split large batch size into micro-batch size.

if this way, i had better prefer to use the batch size to illustrate our perf in the readme, since others do not have any background about the micro batch size and max length. for our tech paper, we can use current way.
